NAudio.Core/Utils/CircularBuffer.cs

```diff
@@ -96,6 +96,47 @@ public int Read(byte[] data, int offset, int count)
             }
         }
 
+        /// <summary>
+        /// Peeks ahead at data in the buffer without advancing the read position.
+        /// </summary>
+        /// <param name="peekOffset">The offset from the current read position where peeking will begin.</param>
+        /// <param name="data">The buffer we will read into.</param>
+        /// <param name="offset">The offset into the <paramref name="data"/> buffer.</param>
+        /// <param name="count">The number of bytes to read.</param>
+        /// <returns>The number of bytes actually read.</returns>
+        public int Peek(uint peekOffset, byte[] data, int offset, int count)
+        {
+            lock (lockObject)
+            {
+                int tmpReadPosition = readPosition + (int)peekOffset;
+                tmpReadPosition %= buffer.Length;
+
+                // The same implementation as Read however we use a local temporary read position variable.
+                if (count > byteCount)
```

Comment:
The implementation of the `Peek` method is actually almost identical to the `Read` method above it. The only difference being that the `Peek` doesn't modify the `readPosition` field, it creates a local temporary variable to log the read position. Also we don't increment the `byteCount` before we return to the caller.